server/middleware.go

```diff
@@ -102,3 +104,60 @@ func withLogging(
 		log.Info("request", args...)
 	}
 }
+
+// withCORS is a middleware that adds CORS headers to responses.
+// It intercepts OPTIONS requests for handling CORS preflight requests.
+func withCORS(
+	handleFn func(http.ResponseWriter, *http.Request),
+	corsConfig config.ServerConfig,
+	log logging.Logger,
+) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// If CORS is not enabled, just call the handler
+		if !corsConfig.CORSEnabled {
+			handleFn(w, r)
+			return
+		}
+
+		origin := r.Header.Get("Origin")
+		if origin == "" {
+			// Not a CORS request
+			handleFn(w, r)
+			return
+		}
+
+		// Check if origin is allowed
+		allowOrigin := "*"
+		if len(corsConfig.CORSAllowedDomains) > 0 {
+			allowed := false
+			for _, domain := range corsConfig.CORSAllowedDomains {
+				if strings.TrimSpace(domain) == origin || 
+				   strings.HasSuffix(origin, strings.TrimSpace(domain)) {
```

Comment:
Seems to have a security flaw
<img width="675" alt="image" src="https://github.com/user-attachments/assets/0ddd726f-5ce9-44ca-91da-07a7e4827c6d" />

Do we really need to match for subdomains? If yes, is there not a golang library that does proper matching of subdomains?
cc @anupsv